pkg/ebpftracer/tracer.go

```diff
@@ -220,7 +225,7 @@ func (t *Tracer) eventsReadLoop(ctx context.Context) error {
 }
 
 func (t *Tracer) runPerfBufReaderLoop(ctx context.Context, target *ebpf.Map) error {
-	eventsReader, err := perf.NewReader(target, t.cfg.EventsPerCPUBuffer)
+	eventsReader, err := ringbuf.NewReader(target)
```

Comment:
In order for the ringbuf to not potentially block forever, we need to spawn a `goroutine` that will invoke the `Close` method on the ringbuf.

Reply:
Something like 
```go
go func() {
		<-ctx.Done()
		eventsReader.Close()
	}()
```